fix: Booker fix for team + user with same event + event slug

JeroenReumkens opened the pull request:
## What does this PR do?

If there would a user that was part of a team that had the exact same slug as the user, AND also both the team and the user have an event with te same slug, it would show the user event on the team page. 

This PR changes that behaviour. Now we have the option in the booker the force it the only fetch a team event. That boolean is passed on the team pages, which results in the query only fetching team events, instead of running an `OR` query and fetching the first match. 

Fixes #8579

## Type of change

- Bug fix (non-breaking change which fixes an issue)

## How should this be tested?

- [ ] Login with team pro, change their profile slug to `seeded`. Go to the seeded team page and change the slug to `seeded` too. Then create an event in both the team and the teampro-profile with a slug `test`. With the changes in this PR, it should simply show both event pages correctly.

JeroenReumkens committed:
#8579: Prevent wrong event from being fetched when both a usser and a team have the same slug AND the same event with same slug

JeroenReumkens commented on apps/web/pages/new-booker/team/[slug]/[type].tsx:
```diff
@@ -30,6 +30,7 @@ export default function Type({ slug, user, booking, away, isBrandingHidden }: Pa
         rescheduleBooking={booking}
         isAway={away}
         hideBranding={isBrandingHidden}
+        isTeamEvent
```
New boolean in booker that can now force it to only fetch team events.

hariombalhara replied:
We would probably have to do it for private links as well. A private link can be a team event or user event.

JeroenReumkens commented on packages/features/bookings/Booker/store.ts:
```diff
@@ -207,9 +217,29 @@ export const useInitializeBookerStore = ({
   useEffect(() => {
-    initializeStore({ username, eventSlug, month, eventId, rescheduleUid, rescheduleBooking, layout });
-  }, [initializeStore, username, eventSlug, month, eventId, rescheduleUid, rescheduleBooking, layout]);
+    initializeStore({
```
Prettier did it's thing. The `isTeamEvent` boolean is added to the effect here.

JeroenReumkens commented on packages/features/bookings/Booker/utils/event.ts:
```diff
@@ -16,9 +16,10 @@ import { useBookerStore } from "../store";
  */
 export const useEvent = () => {
   const [username, eventSlug] = useBookerStore((state) => [state.username, state.eventSlug], shallow);
+  const isTeamEvent = useBookerStore((state) => state.isTeamEvent);
```
the `useEvent` hook takes the `isTeamEvent` boolean from the store and automatically passes that to the query so it fetches the right data.

JeroenReumkens commented on packages/features/eventtypes/lib/getPublicEvent.ts:
```diff
@@ -141,24 +146,34 @@ export const getPublicEvent = async (username: string, eventSlug: string, prisma
     };
   }
 
+  const usersOrTeamQuery = isTeamEvent
```
If it's a team event we now only query for the team, instead of running the OR query. Left the OR in place for all other cases, so team is even then still used as a fallback (for example in case a user in the future will use this as atom, and forget to pass in the `isTeamEvent` boolean.

JeroenReumkens replied:
Yes you are right, I've just now reverted this decision to NOT always fallback to the team event. The `isTeamEvent` boolean is not required though. So if you don't pass it in, it defaults to a user event and won't fetch the team event as fallback anymore.

JeroenReumkens committed:
Set isTeamEvent boolean on private link pages.

hariombalhara commented on packages/features/eventtypes/lib/getPublicEvent.ts:
```diff
@@ -141,24 +146,25 @@ export const getPublicEvent = async (username: string, eventSlug: string, prisma
+          some: {
+            username,
+          },
+        },
```
```suggestion
        },
        team: null,
        
```

hariombalhara replied:
This is needed, otherwise it would also return team events as well and then again we can serve team event instead of user only events.

JeroenReumkens replied:
Wow, this makes a lot of sense yeah! Good spot!

hariombalhara reviewed (changes requested):
Requested one change. Otherwise looks good.
Renamed **seeded-team** slug to **teampro** and then did the following tests
- Visited /new-booker/teampro/30min - Serves user event.
- Visited /new-booker/team/teampro/30min(after creating 30min event in Seeded Team). Serves team event.

JeroenReumkens committed:
Update packages/features/eventtypes/lib/getPublicEvent.ts

Co-authored-by: Hariom Balhara <hariombalhara@gmail.com>

JeroenReumkens committed:
prettier

JeroenReumkens committed:
Merge branch 'main' into fix/prevent-same-slugs-from-breaking-booker

JeroenReumkens commented:
> Requested one change. Otherwise looks good. Renamed **seeded-team** slug to **teampro** and then did the following tests
> 
> * Visited /new-booker/teampro/30min - Serves user event.
> * Visited /new-booker/team/teampro/30min(after creating 30min event in Seeded Team). Serves team event.

Thanks a lot for testing! Just fixed your suggested changing. Re-requesting review 😁
